### PR TITLE
Implement redux

### PR DESCRIPTION
**Pull request by Deftool66**

### CODE REVIEW

File: libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        const email = error.customData.email;
+        const credential = GoogleAuthProvider.credentialFromError(error);
+        console.log('error', error);
+      });
+  };
+  return (
+    <div className={styles.container}>
+      <button onClick={signInWithGoogle}>Sign in with Google</button>
```
Comment (juice1000) [inline libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx:50]: this needs and arrow function 
() => signInWithGoogle

File: libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        const user = result.user;
+        if (user.displayName && user.email && user.photoURL) {
+          const newUser: User = {
+            id: user.uid,
+            email: user.email,
+            name: user.displayName,
+            initials: user.displayName?.slice(0, 2).toUpperCase(),
+            profileImg: user.photoURL,
+          };
+          firestoreService.writeToDB('users', newUser);
```
Comment (juice1000) [inline libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx:36]: nice! next, this needs to be persisted in redux as a state!

File: libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const auth = getAuth();
+export function GoogleLogin() {
+  // creating the sign in with google react component
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
+  provider.addScope('https://www.googleapis.com/auth/contacts.readonly');
+  const signInWithGoogle = () => {
+    console.log('sign in with google');
+    console.log('auth', auth, provider);
```
Comment (juice1000) [inline libs/auth/src/lib/GoogleLogin/GoogleLogin.tsx:20]: for security reasons, the sensitive logs need to be removed

File: apps/toolai/src/firebase-init.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ const config = {
 // Initialize Firebase
 const app = initializeApp(config.firebase);
 // intialize google provider
-export const googleProvider = new GoogleAuthProvider();
+// export const provider = new GoogleAuthProvider();
```
Comment (juice1000) [inline apps/toolai/src/firebase-init.ts:23]: we can remove this line

**Review by juice1000 — CHANGES_REQUESTED** [review]

File: apps/toolai/src/firebase-init.ts (hunk 4 above)
Comment (Deftool66) [reply apps/toolai/src/firebase-init.ts]: yes sir !!

**Review by juice1000 — APPROVED** [review]

**Pull request merged**